mission_systems/shooter/nodes/ShooterControl.py

```diff
@@ -0,0 +1,128 @@
+    dur_from_start = rospy.Duration(0,0)
+    rospy.loginfo("starting load")
+    rate = rospy.Rate(50) # 10hz
+    feedback = ShooterDoActionFeedback()
+    while dur_from_start < self.load_total_time and self.load_server.is_active():
+      dur_from_start = rospy.get_rostime() - start_time
+      feedback.feedback.time_remaining = self.load_total_time - dur_from_start
+      self.load_server.publish_feedback(feedback)
+      if (dur_from_start < self.load_retract_time):
+        self.motor_controller.setMotor1(1.0)
+      elif (dur_from_start < self.load_total_time):
+        self.motor_controller.setMotor1(-1.0)
+      rate.sleep()
+    if(not self.load_server.is_active()):
+      return
+    self.motor_controller.setMotor1(0)
+    self.motor_controller.setMotor2(1)
+    result.result.success = True
+    self.loaded = True
+    self.load_server.set_succeeded(result)
+    rospy.loginfo("Finished loaded")
+  def fireExecuteCallback(self,goal):
+    result = ShooterDoActionResult()
+    if (self.load_server.is_active()):
+      rospy.loginfo("Something already running, aborting")
+      result.result.success = False
+      result.result.error = result.result.ALREADY_RUNNING
+      self.fire_server.set_aborted(result)
+      return    
+    if not self.loaded:
+      rospy.loginfo("Not loaded, aborting fire")
+      result.result.success = False
+      result.result.error = result.result.NOT_LOADED
+      self.fire_server.set_aborted(result)
+      return
+    start_time = rospy.get_rostime()
+    dur_from_start = rospy.Duration(0,0)
+    rospy.loginfo("starting fire")
+    rate = rospy.Rate(50) # 10hz
+    feedback = ShooterDoActionFeedback()
+    while dur_from_start < self.total_fire_time and self.fire_server.is_active():
+      dur_from_start = rospy.get_rostime() - start_time
+      feedback.feedback.time_remaining = self.total_fire_time - dur_from_start
+      self.fire_server.publish_feedback(feedback)
+      if (dur_from_start < self.fire_extend_time):
+        self.motor_controller.setMotor1(-1)
+      else:
+        self.motor_controller.setMotor1(0)
+      rate.sleep()
+    if(not self.fire_server.is_active()):
+      return
+    self.motor_controller.setMotor1(0)
+    self.motor_controller.setMotor2(0)
+    result.result.success = True
+    self.loaded = False
+    self.fire_server.set_succeeded(result)
+    rospy.loginfo("Finished fire")
+  def cancelCallback(self,req):
+    result = ShooterDoActionResult()
+    result.result.success = False
+    result.result.error = result.result.MANUAL_CONTROL_USED
+    if (self.load_server.is_active()):
+      self.load_server.set_aborted(result)
+    if (self.fire_server.is_active()):
+      self.fire_server.set_aborted(result)
+    time.sleep(0.1)
+    self.motor_controller.setMotor1(0)
+    self.motor_controller.setMotor2(0)
+    rospy.loginfo("canceled")
+  def __init__(self):
+    self.controller_file = rospy.get_param('~controller/controller_serial', '/dev/ttyUSB0')
```

Comment:
Make sure this param gets set somewhere in the launch (it looks like it got taken out), else you should probably set the serial port to use the `by-id` path by default.